modules/imgcodecs/src/grfmt_webp.cpp

```diff
@@ -155,7 +144,7 @@ bool WebPDecoder::readData(Mat &img)
     CV_CheckEQ(img.cols, m_width, "");
     CV_CheckEQ(img.rows, m_height, "");
 
-    if (m_buf.empty())
+    if (data.empty())
```

Comment:
That should still be m_buf no ?

Reply:
if (m_buf.empty()) don't pass tests.. m_buf is related with imdecode and imencode which i did not focused yet.

Reply:
@vrabaud i implemented imdecode related part and add related test. could you take a look.